### ISSUE TITLE
Print functions cause const record domain/range disorder with FcnRcdValue

### ISSUE THREAD

**Issue by tangruize** [tlaplus/tlaplus#726]
I am using Print/PrintT functions to debug my specifications in 'Evaluate Constant Expression' section. I found that the Print functions can cause the const record's domain and range disorder. e.g. `[data |-> 15, dst |-> "a"]` becomes `[data |-> "a", dst |-> 15]` after printing it.

Here is the spec to reproduce:
```tla
-------------------------------- MODULE Test --------------------------------

EXTENDS TLC

msgs == [ a |-> [ data |-> 15 ] ]

GetMsg(dst) ==
    LET m == msgs[dst]
    IN [ dst |-> dst] @@ m

PrintPrintGetMsg ==
    LET m == GetMsg("a")  \* expect: [data |-> 15, dst |-> "a"]
    IN <<Print("ok", m) , Print("bad", m)>>

=============================================================================
```

User output:
```txt
"ok"  [data |-> 15, dst |-> "a"]
"bad"  [data |-> "a", dst |-> 15]
```

I don't know if it is a bug or if I did something wrong.

**Comment by lemmy** [comment]
Not a recent regression:

```shell
-> % java -cp /Users/markus/TLC/tla2tools_1-4-0.jar tlc2.TLC Github726
TLC2 Version 2.03 of 20 July 2011
Running in Model-Checking mode.
Parsing file Github726.tla
Parsing file /var/folders/7d/4x6z2cc91jl588ysynlc1tfc0000gn/T/TLC.tla
Parsing file /var/folders/7d/4x6z2cc91jl588ysynlc1tfc0000gn/T/Naturals.tla
Parsing file /var/folders/7d/4x6z2cc91jl588ysynlc1tfc0000gn/T/Sequences.tla
Semantic processing of module Naturals
Semantic processing of module Sequences
Semantic processing of module TLC
Semantic processing of module Github726
Starting... (2022-05-03 08:44:47)
"ok"  [data |-> 15, dst |-> "a"]
"bad"  [data |-> "a", dst |-> 15]
Error: Evaluating assumption line 14, col 8 to line 14, col 23 of module Github726 failed.
TLC expected a boolean value, but did not find one. line 14, col 8 to line 14, col 23 of module Github726
Finished. (2022-05-03 08:44:47)
```

**Comment by lemmy** [comment]
@tangruize Thanks for the report.  Can you please verify with your original, real spec that the issue has been addressed with 95d980c (build will be available at https://nightly.tlapl.us/dist/ after successful completion of https://github.com/tlaplus/tlaplus/runs/6278633007)?

**Comment by tangruize** [comment]
I verified the fix with my original spec. It works as expected! Thank you very much!
